stdlib/public/runtime/CygwinPort.cpp

```diff
@@ -0,0 +1,131 @@
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <windows.h>
+#include <psapi.h>
+#include <mutex>
+
+struct dl_phdr_info
+{
+	void			 *dlpi_addr;
+	const char       *dlpi_name;
+};
+
+int dl_iterate_phdr(int(*callback)(struct dl_phdr_info *info, size_t size, void *data),
+					void *data)
+{
+	DWORD proc_id = GetCurrentProcessId();
+	HANDLE proc_handle = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ,
+									 FALSE, proc_id);
+	if (NULL == proc_handle)
+		return 0;
+
+	int last_ret = 0;
+
+	HMODULE modules[1024];
+	DWORD used_size;
+
+	if (EnumProcessModules(proc_handle, modules, sizeof(modules), &used_size))
+	{
+		for (unsigned int i = 0; i<used_size/sizeof(HMODULE); i++)
+		{
+			char mod_name[MAX_PATH];
+
+			if (GetModuleFileNameExA(proc_handle, modules[i], mod_name,
```

Comment:
What are the conditions under which this call can fail?  If those are obscure system failure cases, let's crash with `swift::fatalError` instead.